agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryClient.java

```diff
@@ -359,17 +358,12 @@ private void populateDefaults(AbstractTelemetryBuilder telemetryBuilder) {
     }
   }
 
-  /** Gets or sets the default instrumentation key for the application. */
-  public void setInstrumentationKey(@Nullable String key) {
-    instrumentationKey = key;
-  }
-
+  @Nullable
   public String getStatsbeatInstrumentationKey() {
-    return statsbeatInstrumentationKey;
-  }
-
-  public void setStatsbeatInstrumentationKey(String key) {
-    statsbeatInstrumentationKey = key;
+    // TODO (trask) I may have broken statsbeat under lazy loading
+    //  (azure functions / azure spring cloud)
+    ConnectionString val = this.connectionString;
+    return val != null ? val.getStatsbeatInstrumentationKey() : null;
```

Comment:
@heyams I'll add a work item to test both statsbeat and quickpulse under lazy loading before 3.3.0 release

Reply:
can we add a todo to remind us?

Reply:
https://msazure.visualstudio.com/One/_sprints/taskboard/TelReachSDKTeam/One/Custom/AEP/CY2022H1%20-%20NickelExtended/Feb?workitem=13279991